designs/multiple-worker-node-groups.md

```diff
@@ -215,12 +214,12 @@ spec:
 For each worker node groups, CAPI spec file will continue to have the following 3 kind fields.
 
 * KubeadmConfigTemplate
-* Machime deployment
+* MachineDeployment
 * VSphereMachineTemplate
 
-For each group, we will append these three fields corresponding to that group in the CAPI spec.
+For each group, we will append these three aforementioned fields corresponding to that group in the CAPI spec.
 
-Right now, the cli assumes that there will be only one group and it treats worker node group configuration array as a collection of only one element. As a result, the controller just refers to the first element of this array in different places of the code. So we need to do the same operations in loops, which includes CAPI spec creation, cluster spec validation etc. Once a CAPI spec is created with this approach, the workload cluster will be created with multiple worker nodes.
+Right now, the cli assumes that there will be only one group and it treats worker node group configuration array as a collection of only one element. As a result, the controller just refers to the first element of this array in different places of the code. So we need to do the same operations in loops, which includes CAPI spec creation, cluster spec validation etc. Once a CAPI spec is created with this approach, the workload cluster will be created with multiple worker nodes. We will use an array of CAPI objects to store the worker node group configurations and then generate CAPI spec file using that array.
```

Comment:
Can you mention that one element of the array of worker node group configurations corresponds to a set of CAPI objects consisting of KubeadmConfig, MachineDeployment, and whatever else is there? Just gives us an understanding as to what to expect, even if there are repeated VSphereMachineConfigs for each of worker node groups.

Reply:
I am not sure, if there is a data structure encompassing all three. But these three types are well defined in capi and capv code bases. What I plan to do is to create a structure of these 3 elements and then create an array of that structure.  I will update the code.